synapse/types/__init__.py

```diff
@@ -883,8 +885,31 @@ async def from_string(cls, store: "DataStore", string: str) -> "StreamToken":
             while len(keys) < len(attr.fields(cls)):
                 # i.e. old token from before receipt_key
                 keys.append("0")
```

Comment:
Will this still work OK? I *think* it is for adding the `un_partial_state_rooms_key`?

Reply:
Good question, but yes it'll work as `"0"` is a valid receipt token (it deserializes to `MultiWriterStreamToken(stream=0)`)

Reply:
Ah, of course! 👍